public/app/features/scenes/core/SceneObjectBase.tsx

```diff
@@ -154,6 +156,18 @@ export abstract class SceneObjectBase<TState extends SceneObjectState = {}> impl
     throw new Error('No data found in scene tree');
   }
 
+  getLayout(): SceneObject<SceneLayoutState> {
+    if (this.constructor.name === 'SceneFlexLayout' || this.constructor.name === 'SceneGridLayout') {
+      return this as SceneObject<SceneLayoutState>;
+    }
+
```

Comment:
@torkelo I'm not super happy with relying on the constructor name. Relying on `instanceof` check introduces circular dependency tho. Think one way of solving this is introducing a property `$isLayout: boolean` that's a marker for layout objects. Any thoughts on this?